app/javascript/components/molecules/Popover.js

```diff
@@ -17,23 +18,21 @@ const useClickOutside = (ref, callback) => {
   })
 }
 
-const Popover = ({ buttonElement, children }) => {
+const Popover = ({ buttonElement, children, onButtonClick }) => {
   const [showPopper, setShowPopper] = useState(false)
+  const [referenceElement, setReferenceElement] = useState(null)
```

Comment:
I know you already have `buttonElement` as the prop but `buttonElementRef` might be more descriptive, since popperElement and arrowElement are also references to elements, right?